pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -490,3 +509,33 @@ func (c *Controller) deletePod(obj interface{}) {
 		c.addPod(pod)
 	}
 }
+
+func (c *Controller) addNode(obj interface{}) {
+	c.checkNodeTopologyDistribution()
+}
+
+func (c *Controller) updateNode(old, cur interface{}) {
+	c.checkNodeTopologyDistribution()
+}
+
+func (c *Controller) deleteNode(obj interface{}) {
+	c.checkNodeTopologyDistribution()
+}
+
+// checkNodeTopologyDistribution updates Nodes in the topology cache and then
+// queues any Services that are past the threshold.
+// TODO(robscott): Explore queue mechanism here.
+func (c *Controller) checkNodeTopologyDistribution() {
+	if c.topologyCache == nil {
+		return
+	}
+	nodes, err := c.nodeLister.List(labels.Everything())
```

Comment:
Yep, on my list to try to optimize. Also asked @wojtek-t about if some kind of queue/batching would help here, but it doesn't sound like that would make a huge difference here, will look into other ways to optimize this.

Reply:
Didn't look in the whole PR, but:
- I mentioned offline to Rob, that we don't want to call that on every update, but only on those that could change something (I guess only Read <-> NotReady transitions?)
- this listing is happening from local cache. What is more this isn't even copying objects:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/listers/core/v1/node.go#L51
 So that one is pretty cheap actually. Rebuilding the cache will be much more expensive

Reply:
Thanks for the extra details @wojtek-t! Does this mean the code is already good enough as is or are there any optimizations I should make?